[24.11] immich: 1.123.0 -> 1.125.6

Scrumplex opened the pull request:
Full backports of

- https://github.com/NixOS/nixpkgs/pull/369042
- https://github.com/NixOS/nixpkgs/pull/371942
- https://github.com/NixOS/nixpkgs/pull/376211
- https://github.com/NixOS/nixpkgs/pull/377084
- https://github.com/NixOS/nixpkgs/pull/377427

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Scrumplex committed:
immich: use Node.js 20

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>
Co-authored-by: Simon Gardling <titaniumtown@proton.me>
(cherry picked from commit d2145a1ad8e92b33f02f21a4290a133129e150f3)

Scrumplex committed:
immich: rely on native nodejs

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>
(cherry picked from commit 7bf88bebfc48455d221ba618f2693cec889de38c)

dotlambda committed:
immich: 1.123.0 -> 1.124.2

Diff: https://github.com/immich-app/immich/compare/refs/tags/v1.123.0...v1.124.2

Changelog: https://github.com/immich-app/immich/releases/tag/v1.124.0
           https://github.com/immich-app/immich/releases/tag/v1.124.1
           https://github.com/immich-app/immich/releases/tag/v1.124.2
(cherry picked from commit b96bce0598349bcf6210db7a5c9275400861843b)

dotlambda committed:
immich: 1.124.2 -> 1.125.2

Diff: https://github.com/immich-app/immich/compare/refs/tags/v1.124.2...v1.125.2

Changelog: https://github.com/immich-app/immich/releases/tag/v1.125.1
           https://github.com/immich-app/immich/releases/tag/v1.125.2
(cherry picked from commit f65d36693a360d31b6d662a9d2d73efc24625135)

dotlambda committed:
immich: 1.125.2 -> 1.125.3

Diff: https://github.com/immich-app/immich/compare/refs/tags/v1.125.2...v1.125.3

Changelog: https://github.com/immich-app/immich/releases/tag/v1.125.3
(cherry picked from commit 0b73be72826763cf13160c250e83568363b031c1)

Scrumplex commented:
> Please cherry pick [d2145a1](https://github.com/NixOS/nixpkgs/commit/d2145a1ad8e92b33f02f21a4290a133129e150f3).

On 24.11, `nodejs` already points to `nodejs_20`. Therefore, that commit wouldn't do anything in 24.11.

Titaniumtown reviewed (approved):
diff LGTM

dotlambda commented:
> > Please cherry pick [d2145a1](https://github.com/NixOS/nixpkgs/commit/d2145a1ad8e92b33f02f21a4290a133129e150f3).
> 
> On 24.11, `nodejs` already points to `nodejs_20`. Therefore, that commit wouldn't do anything in 24.11.

I know, but that would make for clean cherry-picks.

dotlambda committed:
immich: 1.125.3 -> 1.125.6

Diff: https://github.com/immich-app/immich/compare/refs/tags/v1.125.3...v1.125.6

Changelog: https://github.com/immich-app/immich/releases/tag/v1.125.5
           https://github.com/immich-app/immich/releases/tag/v1.125.6
(cherry picked from commit 71ec0e85649a8530d80856cec59272250c9b77af)

diogotcorreia commented:
Perhaps cherry pick https://github.com/NixOS/nixpkgs/pull/377427 as well as https://github.com/NixOS/nixpkgs/pull/377796 (if it gets merged soon)

diogotcorreia reviewed (commented):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 377271`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>immich</li>
    <li>immich-cli</li>
    <li>immich-machine-learning</li>
    <li>immich-machine-learning.dist</li>
  </ul>
</details>
